src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using SFA.DAS.Reservations.Domain.Reservations;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Nest;
+using SFA.DAS.Reservations.Domain.Configuration;
+
+namespace SFA.DAS.Reservations.Data.Repository
+{
+    public class ReservationIndexRepository : IReservationIndexRepository
+    {
+        public const string ReservationIndexLookupName = "-reservations-index-registry";
+
+        private readonly IElasticClient _client;
+        private readonly ReservationsApiEnvironment _environment;
+
+        public ReservationIndexRepository(IElasticClient client, ReservationsApiEnvironment environment)
+        {
+            _client = client;
+            _environment = environment;
+        }
+
+        public async Task<IEnumerable<ReservationIndex>> Find(long providerId, string term)
+        {
+            var searchIndexRegistryResponse = _client.Search<IndexRegistryEntry>(s => s
```

Comment:
Can this be SearchAsync? Additional enhancement to note here might be to cache that reservationIndexName for a day

Reply:
yep good point on the caching, i'll take a look how that can be implemented.  With regards to the async I have a feeling I had trouble with it before but i'll double check and see if I can make the change. May have just been not working due to something else.

Reply:
Can leave the caching for now 😄

Reply:
ok shall we add a story for it in the backlog?